src/main/javascript/crypto/src/libs/IdentifierAttestation.ts

```diff
@@ -1,87 +1,96 @@
+        // this.setSubject("CN=" + this.subjectKey.getAddress());
+        this.setSubject("CN=");
+        this.setSigningAlgorithm(this.DEFAULT_SIGNING_ALGORITHM);
 
         this.setSubjectPublicKeyInfo(keys);
-        this.setRiddle(riddle);
+        this.setCommitment(commitment);
     }
 
     static fromData(identifier: string, type: number, keys: KeyPair, secret: bigint){
         let crypto = new AttestationCrypto();
-        let riddle = crypto.makeCommitment(identifier, type, secret);
-        return new this(riddle, keys);
+        let commitment = crypto.makeCommitment(identifier, type, secret);
+        return (new this()).fromCommitment(commitment, keys);
+        //return new this(riddle, keys);
     }
 
-    setSignature(signature: string) {
-        // TODO create algorithm parser and change variable type
-        // let alg = new AlgorithmIdentifierASN();
-        // alg.algorithm = signature;
-        // hadrcoded ASN1 sequence
-        this.signature = "300906072A8648CE3D0201";
+    static fromBytes(bytes: Uint8Array){
+        return super.fromBytes(bytes);
     }
-    setSubjectPublicKeyInfo(keys: KeyPair){
-        // TODO generate algorithm object
-        // hardcoded algorithm
-        // let hardcodedAlg = "3081EC06072A8648CE3D02013081E0020101302C06072A8648CE3D0101022100FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFFC2F3044042000000000000000000000000000000000000000000000000000000000000000000420000000000000000000000000000000000000000000000000000000000000000704410479BE667EF9DCBBAC55A06295CE870B07029BFCDB2DCE28D959F2815B16F81798483ADA7726A3C4655DA4FBFC0E1108A8FD17B448A68554199C47D08FFB10D4B8022100FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141020101";
 
-        // let res = hardcodedAlg + Asn1Der.encode('BIT_STRING', keys.getPublicKeyAsHexStr());
-        // this.subjectPublicKeyInfo = Asn1Der.encode('SEQUENCE_30', res);
-        this.subjectPublicKey = keys;
+    setSubjectPublicKeyInfo(keys: KeyPair){
+        this.subjectKey = keys;
     }
 
-
-    setRiddle(riddle: Uint8Array) {
-        // TODO hardcoded OID
-        // extensions.add(new ASN1ObjectIdentifier(Attestation.OID_OCTETSTRING));
-        let attestOIDencoded = "060B2B060104018B3A73790128";
-        let extensions: string = attestOIDencoded
-            + Asn1Der.encode('BOOLEAN', 1)
-        + Asn1Der.encode('OCTET_STRING', uint8tohex(riddle));
-
-        let extensionsEncoded = Asn1Der.encode('SEQUENCE', extensions);
-        extensionsEncoded = Asn1Der.encode('SEQUENCE', extensionsEncoded);
-
-        this.riddle = riddle;
-        // Double Sequence is needed to be compatible with X509V3
-        // TODO create extensions as correct type "Extensions"
-        // this.extensions = extensionsEncoded;
+    setCommitment(encodedRiddle: Uint8Array) {
+        this.commitment = encodedRiddle;
     }
 
     checkValidity(): boolean {
         if (!super.checkValidity()) {
             return false;
```

Comment:
This still seems to require some update according to the java implementation

Reply:
class updated except of RDNs, I have to figure out how to do it in JS correctly. We can create issue for that and I will update it later.